components/email-mgt/org.wso2.carbon.email.mgt/src/test/java/org/wso2/carbon/email/mgt/OrganizationEmailTemplateTest.java

```diff
@@ -406,6 +406,7 @@ private void mockRegistryResource(String notificationChannel, String displayName
 
         when(resourceMgtService.getIdentityResource(Matchers.anyString(), Matchers.anyString(), Matchers.anyString()))
                 .thenReturn(resource);
+        when(resourceMgtService.isResourceExists(Matchers.anyString(), Matchers.anyString())).thenReturn(true);
```

Comment:
Why we needed to mock this now?

Reply:
Due to the new if condition we have at [1]. Previously we did not check if it exists.
We need [2] to return true from the mocked registry resource. 

[1] https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/files#diff-55202b63368a9acaf49ac62f2c542327e0302c40fd60374d7378a6dcaaafd511R58-R60
[2] https://github.com/wso2-extensions/identity-event-handler-notification/blob/d610d0c389e5674ff60065a879f91d3835bb1210/components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/RegistryBasedTemplateManager.java#L91

Reply:
I'm still not clear, what the reason for this..

> Due to the new if condition we have at [1]. Previously we did not check if it exists.

You are implying the implementation of the `isNotificationTemplateTypeExists()` method was the root cause, but I think the root case don't have to do with the method implementation of `isNotificationTemplateTypeExists()` method.

I think, we have introduced a new code elsewhere, that caused for a new invocation of 'isNotificationTemplateTypeExists()` that wasn't there earlier..

By definition, if `isNotificationTemplateTypeExists()` method get called, it should have check for existence of the resource, so this mock should be there already..

Since we have to introduce this mock with this change means, we are now doing additional isExists() check in a flow elsewhere.. wanted to clear that this additional check is intentional.

Reply:
Previously provided point of reference was wrong. This is due to the new if condition we have at [1]

[1] https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/files#diff-55202b63368a9acaf49ac62f2c542327e0302c40fd60374d7378a6dcaaafd511R109

Reply:
We have removed this if condition with https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/commits/8628c42d5f74b1e58a734be7fc641c3f846af46d.
So shall we remove this additional mock?

Reply:
Ack, yes. Removed.